packages/babel-core/src/config/files/module-types.ts

```diff
@@ -87,21 +87,43 @@ function loadCtsDefault(filepath: string) {
     handler = function (m, filename) {
       // If we want to support `.ts`, `.d.ts` must be handled specially.
       if (handler && filename.endsWith(ext)) {
-        // @ts-expect-error Undocumented API
-        return m._compile(
-          transformFileSync(filename, {
-            ...opts,
+        try {
+          // @ts-expect-error Undocumented API
+          return m._compile(
+            transformFileSync(filename, {
+              ...opts,
+              filename,
+            }).code,
             filename,
-          }).code,
-          filename,
-        );
+          );
+        } catch (error) {
+          if (!hasTsSupport) {
+            const packageJson = require("@babel/preset-typescript/package.json");
+
+            if (
+              semver.lte(
+                // eslint-disable-next-line import/no-extraneous-dependencies
+                packageJson.version,
+                "7.21.0",
```

Comment:
We should change this before merging.

Reply:
Before releasing, right? Could you add a notice in `new-version-check` in the makefile?